crepecake/src/main/java/com/google/cloud/tools/crepecake/registry/RegistryClient.java

```diff
@@ -74,7 +87,36 @@ public void pushManifest(V22ManifestTemplate manifestTemplate, String imageTag)
   public Blob pullBlob(DescriptorDigest blobDigest, Path destPath)
       throws RegistryException, IOException {
     BlobPuller blobPuller = new BlobPuller(registryEndpointProperties, blobDigest, destPath);
-    return callRegistryEndpoint(null, blobPuller);
+    return callRegistryEndpoint(blobPuller);
+  }
+
+  // TODO: Add mount with 'from' parameter
+  /**
+   * Pushes the BLOB, or skips if the BLOB already exists on the registry.
+   *
+   * @param blobDigest the digest of the BLOB, used for existence-check
+   * @param blob the BLOB to push
+   * @return {@code true} if the BLOB already exists on the registry and pushing was skipped; false
+   *     if the BLOB was pushed
+   */
+  public boolean pushBlob(DescriptorDigest blobDigest, Blob blob)
+      throws IOException, RegistryException {
+    BlobPusher blobPusher = new BlobPusher(registryEndpointProperties, blobDigest, blob);
+
+    // POST /v2/<name>/blobs/uploads/?mount={blob.digest}
+    String locationString = callRegistryEndpoint(blobPusher.initializer());
+    if (locationString == null) {
+      return true;
+    }
+    URL location = new URL(locationString);
+
+    // PATCH <Location> with BLOB
```

Comment:
can you elaborate on this comment a little? What is that new "location" you're  getting back from this call?

Reply:
I added some comments to BlobPusher about the process. Essentially, the push is conducted in three stages:

1) POST - initializes the upload - gets a Location back to send the BLOB content to
2) PATCH - write the BLOB to the upload Location, gets a Location back to commit the BLOB digest
3) PUT - commit the BLOB to a specific digest id